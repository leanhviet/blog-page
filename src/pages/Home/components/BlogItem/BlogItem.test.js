// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { BlogItemBase } from './index'

const props = {
  blog: {
    id: 1,
    content: 'This is content',
    title: 'Title',
  },
}

describe('BlogItemBase component', () => {
  const wrapper = shallow(<BlogItemBase />)
  const wrapperWithProps = shallow(<BlogItemBase {...props} />)

  it('should be defined BlogItemBase component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot BlogItemBase component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps BlogItemBase component', () => {
    expect(BlogItemBase.defaultProps.blog).toEqual({})
  })

  // Testing mock props
  it('should be render mockProps DetailBase component', () => {
    expect(wrapperWithProps.find('Link').props().children).toEqual(
      props.blog.title,
    )
    expect(wrapperWithProps.find('p').props().children).toBe(props.blog.content)
  })
})
