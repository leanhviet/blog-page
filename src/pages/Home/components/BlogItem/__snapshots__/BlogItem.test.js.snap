// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItemBase component should be render snapshot BlogItemBase component 1`] = `
<Row
  className="blog-item"
  noGutters={false}
>
  <Col
    className="blog-item__wrapper-image"
    sm={2}
    xs={5}
  >
    <img
      className="img-thumbnail"
      src="https://dummyimage.com/150x150/5a5a5a/ffffff"
    />
  </Col>
  <Col
    className="blog-item__content"
    sm={10}
    xs={7}
  >
    <h4
      className="text-primary"
    >
      <Link
        className="link"
        to="/undefined"
      />
    </h4>
    <p />
  </Col>
</Row>
`;

exports[`BlogItemBase component should be render snapshot BlogItemBase component 2`] = `
<Row
  className="blog-item"
  noGutters={false}
>
  <Col
    className="blog-item__wrapper-image"
    sm={2}
    xs={5}
  >
    <img
      alt="Title"
      className="img-thumbnail"
      src="https://dummyimage.com/150x150/5a5a5a/ffffff"
    />
  </Col>
  <Col
    className="blog-item__content"
    sm={10}
    xs={7}
  >
    <h4
      className="text-primary"
    >
      <Link
        className="link"
        to="/1"
      >
        Title
      </Link>
    </h4>
    <p>
      This is content
    </p>
  </Col>
</Row>
`;
